jina/peapods/runtimes/asyncio/rest/models.py

```diff
@@ -1,11 +1,153 @@
+    RouteProto,
+    EnvelopeProto,
+    StatusProto,
+    MessageProto,
+    RequestProto,
+    QueryLangProto,
+)
+
+
+DEFAULT_REQUEST_SIZE = set_client_cli_parser().parse_args([]).request_size
+PROTO_TO_PYDANTIC_MODELS = {}
+PROTOBUF_TO_PYTHON_TYPE = {
+    FieldDescriptor.TYPE_INT32: int,
+    FieldDescriptor.TYPE_INT64: int,
+    FieldDescriptor.TYPE_UINT32: int,
+    FieldDescriptor.TYPE_UINT64: int,
+    FieldDescriptor.TYPE_SINT32: int,
+    FieldDescriptor.TYPE_SINT64: int,
+    FieldDescriptor.TYPE_BOOL: bool,
+    FieldDescriptor.TYPE_FLOAT: float,
+    FieldDescriptor.TYPE_DOUBLE: float,
+    FieldDescriptor.TYPE_FIXED32: float,
+    FieldDescriptor.TYPE_FIXED64: float,
+    FieldDescriptor.TYPE_SFIXED32: float,
+    FieldDescriptor.TYPE_SFIXED64: float,
+    FieldDescriptor.TYPE_BYTES: bytes,
+    FieldDescriptor.TYPE_STRING: str,
+    FieldDescriptor.TYPE_ENUM: Enum,
+    FieldDescriptor.TYPE_MESSAGE: None,
+}
+
+
+def protobuf_to_pydantic_model(
+    protobuf_model: Union[Descriptor, GeneratedProtocolMessageType]
+) -> BaseModel:
+    """
+    Converts Protobuf messages to Pydantic model for jsonschema creation/validattion
+
+    ..note:: Model gets assigned in the global dict :data:PROTO_TO_PYDANTIC_MODELS
+
+    :param protobuf_model: *Proto message from proto file
+    :type protobuf_model: Union[Descriptor, GeneratedProtocolMessageType]
+    :return: Pydantic model
+    :rtype: BaseModel
+    """
+
+    all_fields = {}
+    oneof_fields = defaultdict(list)
+
+    if isinstance(protobuf_model, Descriptor):
+        model_name = protobuf_model.name
+        protobuf_fields = protobuf_model.fields
+    elif isinstance(protobuf_model, GeneratedProtocolMessageType):
+        model_name = protobuf_model.DESCRIPTOR.name
+        protobuf_fields = protobuf_model.DESCRIPTOR.fields
+
+    if model_name.endswith('Proto') and model_name in PROTO_TO_PYDANTIC_MODELS:
+        return PROTO_TO_PYDANTIC_MODELS[model_name]
+
+    for f in protobuf_fields:
+        field_type = PROTOBUF_TO_PYTHON_TYPE[f.type]
+        default_value = f.default_value
+
+        if f.containing_oneof:
+            # Proto Field type: oneof
+            # NOTE: oneof fields are handled as a post-processing step
+            oneof_fields[f.containing_oneof.name].append(f.name)
+
+        if field_type is Enum:
+            # Proto Field Type: enum
+            enum_dict = {}
+            for enum_field in f.enum_type.values:
+                enum_dict[enum_field.name] = enum_field.number
+            field_type = Enum(f.enum_type.name, enum_dict)
+
+        if f.message_type:
+
+            if f.message_type.name == 'Struct':
+                # Proto Field Type: google.protobuf.Struct
+                field_type = Dict
+                default_value = {}
+
+            elif f.message_type.name == 'Timestamp':
+                # Proto Field Type: google.protobuf.Timestamp
+                field_type = datetime
+                default_value = datetime.now()
```

Comment:
Isn't it better to put an stupid default value? Like this we can see if the timestamp is real or not? Normally timestamp is needed to measure some stuff about the presence of an object in a Pod, like this we try the "false positives" of seeing a meaningful timestamp

Reply:
`Timestamp` field needs further polishing. `google.protobuf.Timestamp` describes time in `seconds` & `nanos`, where `datetime.now()` is in a different format. Since this schema is only exposed to the users for viewing, for now, I would prefer to keep it as-is for now and upgrade as and when required.